shared/src/main/scala/mlscript/JSBackend.scala

```diff
@@ -532,130 +602,142 @@ class JSBackend(allowUnresolvedSymbols: Boolean) {
           case Some(_) => throw CodeGenError("can not have more than one parents.")
           case _ => throw CodeGenError(s"unresolved symbol in parents: $res")
         }
+        case (S(res), S(f: JSField)) => scope.resolveValue(f.property.name) match {
+          case Some(CapturedSymbol(out, sym: MixinSymbol)) =>
+            S(JSInvoke(translateCapcture(CapturedSymbol(out, sym)), Ls(res)))
+          case Some(_) => throw CodeGenError("can not have more than one parents.")
+          case _ => throw CodeGenError(s"unresolved symbol in parents: $res")
+        }
+        case (S(res), S(JSInvoke(f: JSField, _))) => scope.resolveValue(f.property.name) match {
+          case Some(CapturedSymbol(out, sym: MixinSymbol)) =>
+            S(JSInvoke(translateCapcture(CapturedSymbol(out, sym)), Ls(res)))
+          case Some(_) => throw CodeGenError("can not have more than one parents.")
+          case _ => throw CodeGenError(s"unresolved symbol in parents: $res")
+        }
```

Comment:
There's a huge amount of duplicated code here. Please organize it in a smarter way so as not to repeat yourself, using local helper `def`s when needed.

Reply:
Refactored by creating a helper function.